Multi form types for a single widget phase 2 & 3

juzen2003 opened the pull request:
- Fixes multi form types for a single widget phase 2 & 3, and #1267 
- Were any Django, import pipeline, or table_schema files modified? Y
  - If YES:
    - Database used: re-import
    - All Django tests pass: Y
    - FLAKE8 run on modified code: Y
- Were any JavaScript or CSS files modified? Y
  - If YES:
    - JSHINT run on all affected files: Y
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: Y
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA
- Was the documentation reviewed for necessary changes or additions? NA
  - About
  - Getting Started
  - FAQ
  - API Guide
  - Tooltips

Description of changes:
- Update _create_mult to:
  - take in disp as a new parameter that will be used to hide/show a mult option.
  - take in aliases as a new parameter that will store a list of aliases for a mult option.
- Add aliases field to mult table (except ones with preprogrammed mult options)
- Add a check mark next to the category that has any children checkboxes checked.

Known problems:

juzen2003 committed:
Update _create_mult & update_mult_table to take in one more parameter "disp"
to determine if the field should be displayed in the search widget. Default
is set to 'Y', and the field will be hidden if 'N' is stored in the database.

juzen2003 committed:
Add aliases field to mult table, it will store a json list of string,
and default is set to null.
- Update _create_mult, update_mult_table, _convert_sql_response_to_mult_table
  and _mult_table_column_names to properly update the value for aliases
  field in the mult table.

juzen2003 committed:
Add addCheckMarkForCategories to put a check mark as an indication of
which categories have boxes checked.

juzen2003 committed:
Add margin to fine tune the location of checked mark next to the categories.

juzen2003 committed:
Update comments for _convert_sql_response_to_mult_table

juzen2003 committed:
Fix code style in search.js by wrapping code with {} for a if block.

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -2185,6 +2185,10 @@ even when screen is narrow */
     font-weight:bold;
 }
 
+.op-checked-indication {
```
Should this maybe be something like `op-mult-group-checked-indication`?

rfrenchseti commented on opus/application/static_media/js/search.js:
```diff
@@ -529,6 +533,36 @@ var o_search = {
+        let categories = $(`#widget__${slug} .mult_group`);
+        // Check each checkboxes under each category. If a category
+        // has any checked checkbox, add a check mark next to the
+        // category name
```
```suggestion
        // Check each checkbox under each category. If a category
        // has any checked checkboxes, add a check mark next to the
        // category name
```

rfrenchseti commented on opus/application/static_media/js/search.js:
```diff
@@ -529,6 +533,36 @@ var o_search = {
+            let checkboxes = $(cat).find("input.multichoice");
+            for (let checkbox of checkboxes) {
+                isChecked = $(checkbox).is(":checked");
+                if (isChecked) {break;}
```
Put break on separate line? Also are the {} necessary?

rfrenchseti commented on opus/application/static_media/js/search.js:
```diff
@@ -529,6 +533,36 @@ var o_search = {
+                                "<i class='fa fa-check'></i></span>";
+                $(`.${parentCatClassName}`).append(checkMark);
+            } else if (!isChecked) {
+                // Remove the check mark if none of checkboxes under that category is checked.
```
```suggestion
                // Remove the check mark if none of checkboxes under that category are checked.
```

rfrenchseti reviewed (changes requested):
This bug was probably introduced in the sub/sub/category pull, because it isn't in production, but please go ahead and fix it here. If you have a selected item in a category and refresh the browser, the category opens as expected, but the +/- indicator is now wrong.

juzen2003 committed:
Fixed the expand/collapse (+/-) icons for mult groups when the search
tab is first loaded.
- In widgets.js (line 981-994), update expandInputGroupBySelectedTarget
  to properly toggle between "+" & "-" based on the selected values when
  the search tab is first loaded.

juzen2003 committed:
Increase the indentation of subcategories for mult

juzen2003 committed:
Make indentation for subcategories consistent across different browsers.

rfrenchseti commented on opus/application/static_media/css/opus.css:
```diff
@@ -2205,6 +2209,11 @@ grouping icon with non-grouping checkboxes */
 }
 
+/* Increase the indentation of subcategories for mult */
+.mult_group {
```
Didn't we standardize on using "-" instead of "_"? So should be .mult-group

juzen2003 committed:
Rename mult_group* to mult-group*, mult_group_* to mult-group-*, and
mult_group_label_* to mult-group-label-*
